src/main/java/akka/cluster/bootstrap/demo/ClusterApp.java

```diff
@@ -16,63 +20,48 @@
 import akka.management.cluster.bootstrap.ClusterBootstrap;
+import akka.management.javadsl.AkkaManagement;
 import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
 
 import java.util.HashSet;
 import java.util.Set;
 import java.util.concurrent.CompletionStage;
 
 public class ClusterApp extends AllDirectives {
 
-  ClusterApp() {
-    ActorSystem system = ActorSystem.create();
-
-    final Http http = Http.get(system);
-    Materializer materializer = ActorMaterializer.create(system);
-    Cluster cluster = Cluster.get(system);
-
-    system.log().info("Started [" + system + "], cluster.selfAddress = " + cluster.selfAddress() + ")");
-
-    AkkaManagement.get(system).start();
-    ClusterBootstrap.get(system).start();
-
-    cluster
-      .subscribe(system.actorOf(Props.create(ClusterWatcher.class)), ClusterEvent.initialStateAsEvents(), ClusterEvent.ClusterDomainEvent.class);
-
-    // create actors
-    ActorRef noisyActor = createNoisyActor(system);
-
-    final Flow<HttpRequest, HttpResponse, NotUsed> routeFlow = createRoutes(system, cluster).flow(system, materializer);
-    final CompletionStage<ServerBinding> binding = http.bindAndHandle(routeFlow,
-            ConnectHttp.toHost("0.0.0.0", 8080), materializer);
-
-    cluster.registerOnMemberUp(() -> {
-      system.log().info("Cluster member is up!");
-      noisyActor.tell(PoisonPill.getInstance(), ActorRef.noSender());
-    });
-
+  private ClusterApp() {
+    ActorSystem system = ActorSystem.create(Behaviors.setup(context -> {
+        akka.actor.ActorSystem classicSystem = Adapter.toClassic(context.getSystem());
+        Cluster cluster = Cluster.get(context.getSystem());
+        context.getLog().info("Started [" + context.getSystem() + "], cluster.selfAddress = " + cluster.selfMember().address() + ")");
+        final Http http = Http.get(classicSystem);
+        Materializer materializer = Materializer.createMaterializer(classicSystem);
+        final Flow<HttpRequest, HttpResponse, NotUsed> routeFlow = createRoutes(classicSystem, cluster).flow(classicSystem, materializer);
+        final CompletionStage<ServerBinding> binding = http.bindAndHandle(routeFlow,
+                ConnectHttp.toHost("0.0.0.0", 8080), materializer);
+
+        context.spawn(NoisyActor.create(), "NoisyActor");
+        ActorRef<ClusterEvent.MemberEvent> clusterWatcher = context.spawn(ClusterWatcher.create(), "ClusterWatcher");
+
+        cluster.subscriptions().tell(new Subscribe<>(clusterWatcher, ClusterEvent.MemberEvent.class));
+
+        AkkaManagement.get(classicSystem).start();
+        ClusterBootstrap.get(classicSystem).start();
+        return Behaviors.empty();
+    }),"ClusterApp");
   }
 
-  private ActorRef createNoisyActor(ActorSystem system) {
-      return system.actorOf(NoisyActor.props(), "NoisyActor");
-  }
 
-  private Route createRoutes(ActorSystem system, Cluster cluster) {
+  private Route createRoutes(akka.actor.ActorSystem system, Cluster cluster) {
```

Comment:
Looks like we only pass this in to get the logger... perhaps we should pass in the logger?